R/read_cargo_metadata.R

```diff
@@ -25,22 +29,44 @@
 #' read_cargo_metadata()
 #' }
 #'
-read_cargo_metadata <- function(path = ".") {
+read_cargo_metadata <- function(
+    path = ".",
+    dependencies = FALSE,
+    echo = TRUE) {
   check_string(path, class = "rextendr_error")
+  check_bool(dependencies, class = "rextendr_error")
+  check_bool(echo, class = "rextendr_error")
 
-  root <- rprojroot::find_package_root_file(path = path)
+  rust_folder <- rprojroot::find_package_root_file(
+    "src", "rust",
+    path = path
+  )
 
-  rust_folder <- normalizePath(
-    file.path(root, "src", "rust"),
-    winslash = "/",
-    mustWork = FALSE
+  args <- c(
+    "metadata",
+    "--format-version=1",
+    if (isFALSE(dependencies)) {
+      "--no-deps"
+    },
+    if (tty_has_colors()) {
+      "--color=always"
+    } else {
+      "--color=never"
+    }
   )
 
   out <- processx::run(
```

Comment:
Why not extract calls to `processx::run()` together with error handling and input validation and reuse it everywhere?

Reply:
i've added a draft of a `run_cargo()` function that handles all `processx::run()` logic